cmd/gitserver/main.go

```diff
@@ -50,10 +56,26 @@ func main() {
 	if err != nil {
 		log.Fatalf("parsing $SRC_REPOS_DESIRED_PERCENT_FREE: %v", err)
 	}
+
+	repoStore, err := getRepoStore()
+	if err != nil {
+		log.Fatalf("failed to initialize db repo store: %v", err)
+	}
+
 	gitserver := server.Server{
 		ReposDir:                reposDir,
 		DeleteStaleRepositories: runRepoCleanup,
 		DesiredPercentFree:      wantPctFree2,
+		GetRemoteURL: func(ctx context.Context, repo api.RepoName) (string, error) {
+			r, err := repoStore.GetByName(ctx, repo)
+			if err != nil {
+				return "", err
+			}
+			for _, info := range r.Sources {
+				return info.CloneURL, nil
+			}
+			return "", fmt.Errorf("internal error no sources for %s", repo)
```

Comment:
Why do we need to say `internal error`?

```suggestion
			return "", fmt.Errorf("no sources for %q", repo)
```